java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToVarHint.java

```diff
@@ -148,9 +173,16 @@ private static boolean preConditionChecker(HintContext ctx) {
         }
 
         TreePath treePath = ctx.getPath();
+        EnhancedForLoopTree efl = null;
+        TypeMirror expressionType = null;
+        if (treePath.getLeaf() instanceof EnhancedForLoopTree) {
+            efl = (EnhancedForLoopTree) treePath.getLeaf();
+            expressionType = ctx.getInfo().getTrees().getTypeMirror(new TreePath(treePath, efl.getExpression()));
+        }
 
         // variable should have local scope
-        if (info.getTrees().getElement(treePath).getKind() != ElementKind.LOCAL_VARIABLE) {
+        //No need to DeclaredType check as primitive types can also be replaced with var in enhanced-for-loop
+        if ((info.getTrees().getElement(treePath) != null && info.getTrees().getElement(treePath).getKind() != ElementKind.LOCAL_VARIABLE) && expressionType == null) {
```

Comment:
cache `info.getTrees().getElement(treePath)` in a local variable. I didn't get the exact meaning of the condition - is it intended, that the processing continues if e.g. `.getKind() != ELementKind.LOCAL_VARIABLE` provided that `expressionType != null` ?

Reply:
Yes. in case of an assignment operation, the var hint should be present only if it a local variable assignment. The check on ```expressionType == null``` is for a variable in an enhanced for loop, as the iterating variable there would automatically have a local scope.